packages/fxa-auth-server/test/oauth/routes/token.js

```diff
@@ -346,10 +369,116 @@ describe('/token POST', function () {
         emitMetricsEvent: () => {},
       };
       await route.config.handler(request);
-      sinon.assert.calledOnceWithExactly(
-        mockGlean.oauth.tokenCreated,
-        request,
-        { uid: UID, oauthClientId: CLIENT_ID, reason: 'authorization_code' }
+      await new Promise((resolve) =>
+        setTimeout(() => {
+          sinon.assert.calledOnceWithExactly(
+            mockGlean.oauth.tokenCreated,
+            request,
+            {
+              uid: UID,
+              oauthClientId: CLIENT_ID,
+              reason: 'authorization_code',
+              syncDeviceId: '42cbcdce5c0a84a532b2722633af9f03',
```

Comment:
Should this be `mockDevice.findDeviceIdByUidAndSessionId`?

https://github.com/mozilla/fxa/blob/d8441c6cfe6a5bfd7fd7ede6dac3cf483e9f634e/packages/fxa-auth-server/test/oauth/routes/token.js#L28-L29

Reply:
No, the return value is used not the function itself.